Bottom: ce7954e341082ec5c8249e494d1f33a77dff6375
Top:    00a3e6a600b745f520ff519cf07ba63c512a74b2
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-05 16:11:37 +0100

hid/gtk: Use gtk_notebook_get_n_pages() rather than diving in and counting

In GTK3.0, direct access to notebook->children will be impossible.


---

diff --git a/src/hid/gtk/gui-config.c b/src/hid/gtk/gui-config.c
index 851571a..698f57a 100644
--- a/src/hid/gtk/gui-config.c
+++ b/src/hid/gtk/gui-config.c
@@ -1980,7 +1980,7 @@ config_page_create (GtkTreeStore * tree, GtkTreeIter * iter,
 
   vbox = gtk_vbox_new (FALSE, 0);
   gtk_notebook_append_page (notebook, vbox, NULL);
-  page = g_list_length (notebook->children) - 1;
+  page = gtk_notebook_get_n_pages (notebook) - 1;
   gtk_tree_store_set (tree, iter, CONFIG_PAGE_COLUMN, page, -1);
   return vbox;
 }
